src/ConcreteOperations/intersection.jl

```diff
@@ -2,6 +2,47 @@
 
 export intersection
 
+for T in [:LazySet, :AbstractSingleton]
+    @eval begin
+        function intersection(∅::EmptySet, X::$T)
+            @assert dim(∅) == dim(X) "cannot take the intersection between a " *
+                "$(dim(∅))-dimensional empty set and a $(dim(X))-dimensional set"
+            return ∅
+        end
+
+        # symmetric method
+        function intersection(X::$T, ∅::EmptySet)
```

Comment:
this could make use of `@commutative`  once we have it merged in master https://github.com/JuliaReach/LazySets.jl/pull/2286